TemplePlus/inventory.cpp

```diff
@@ -1355,8 +1363,51 @@ void InventorySystem::ItemPlaceInIdx(objHndl item, int idx)
 
 int InventorySystem::GetAppraisedWorth(objHndl item, objHndl appraiser, objHndl vendor, SkillEnum skillEnum)
 {
-	auto getAppraisedWorth = temple::GetRef<int(__cdecl)(objHndl, objHndl, objHndl, SkillEnum)>(0x10069970);
-	return getAppraisedWorth(item, appraiser, vendor, skillEnum);
+	auto skillLevel = dispatch.dispatch1ESkillLevel(appraiser, skillEnum, nullptr, objHndl::null, 1);

```

Comment:
Let me check. Thanks!

Reply:
I checked this for non-prof skill - my Wizard. She has 4.5 rank and 4 INT mod = 8.5 This dispatch1ESkillLevel function will return 8 for her, which is correct.

toee.dll has not clear code SkillLevelGet_1007D430 used there:
```
double __cdecl SkillLevelGet_1007D430(ObjHndl objHnd, int skillIdx)
{
  signed int skillRanks; // ebp@1
  float skillLevelDispatch; // ST50_4@1
  float v4; // ST50_4@1
  BonusList bonlist; // [sp+18h] [bp-378h]@1

  initBonusList(&bonlist);
  skillRanks = getArrayFieldInt32(objHnd, obj_f_critter_skill_idx, skillIdx);
  skillLevelDispatch = (double)Dispatch30SkillLevelGet(objHnd, (SkillEnum)skillIdx, &bonlist, 0i64, 1);
  v4 = (double)(signed int)getArrayFieldInt32(objHnd, obj_f_critter_skill_idx, skillIdx) * 0.5
     + skillLevelDispatch
     - (double)(skillRanks / 2);
  nullsub_1(&bonlist);
  return v4;
}
```
I guess this is to get 8.5, not 8 as Dispatch30SkillLevelGet would otherwise return.

Reply:
Please use the original function.

Reply:
Fixed and tested. Now it handles 8.5 value in my previous example.